sdk/iot/provisioning/src/az_iot_provisioning_client_sas.c

```diff
@@ -142,17 +128,24 @@ AZ_NODISCARD az_result az_iot_provisioning_client_sas_get_password(
   if (az_span_size(key_name) > 0)
   {
     // Key Name
+    AZ_RETURN_IF_NOT_ENOUGH_SIZE(mqtt_password_span,
+      1 // AMPERSAND
+      + az_span_size(skn_string) 
+      + 1 // EQUAL_SIGN
+      + az_span_size(key_name));
+
     mqtt_password_span = az_span_copy_u8(mqtt_password_span, AMPERSAND);
     mqtt_password_span = az_span_copy(mqtt_password_span, skn_string);
     mqtt_password_span = az_span_copy_u8(mqtt_password_span, EQUAL_SIGN);
     mqtt_password_span = az_span_copy(mqtt_password_span, key_name);
   }
 
+  AZ_RETURN_IF_NOT_ENOUGH_SIZE(mqtt_password_span, 1 /* NULL TERMINATOR */);
   mqtt_password_span = az_span_copy_u8(mqtt_password_span, STRING_NULL_TERMINATOR);
 
   if (out_mqtt_password_length != NULL)
   {
-    *out_mqtt_password_length = ((size_t)required_size - 1);
+    *out_mqtt_password_length = ((size_t)mqtt_password_size - (size_t)az_span_size(mqtt_password_span) - 1);
```

Comment:
comment for `NULL` terminator / call to `sizeof()` for subtraction.

Reply:
Added comment.